### DISCUSSION TITLE
UHC In Network File Structure

### DISCUSSION THREAD

**Discussion by lkoll** [CMSgov/price-transparency-guide#572] [Q&A]
I'm trying to sift through the In Network files for UHC. It seems that they have broken their json file into ~10 files (there seems to be some variance to this across different json files). List of sub files within each json below.

- ALLDATA-Seems to contain a bunch of data in long format, but also seems unusable because there's no record identifier for transposing
- IN_NETWORK
- NEGOTIATED_PRICES_SERVICE_CODE
- NEGOTIATEED_RATES_NEGOTIATED_PR
- NEGOTIATED_RATES_PROVIDER_REFE
- PROVIDER_GROUPS_NPI
- PROVIDER_GROUPS_TIN
- PROVIDER_REFERENCES
- ROOT-one line, just seems to be saying it is for "UHC"
- WIDE_ALLDATA-No data

Besides the three files I have notes by above, the other 7 seem to be a relational structure. It seems as though you can connect In_Network to Provider_References to Negotiated_Rates_Provider_Refe to Negotiated_Rates_Negotiated_Pr to Negotiated_Prices_Service_Code

But this leaves out the two files for Provider_Groups_NPI and Provider_Groups_TIN. While Provider_References contains a column called provider_group_id and ordinal_provider_references, neither of these seem to be able to connect to the TIN and NPI tables (Provider_Group_Id in Provider_References goes up to ~25k, ordinal_provider_groups in either Provider_Groups_TIN or Provider_Groups_NPI goes up to ~60k). Is there a schema laying out how these connect for UHC specifically? Or one laying it out for all providers that gets at these questions I have? After looking under the schema folder it seemed like UHC was somewhat breaking from the variable and file name structures listed in there, but if I've missed something please let me know.

Edit: I guess I should mention I have been using SAS to read data out of the json files with their json functionality.

Edit 2: One note on above that isn't critical to my question, the ALLDATA table that is created from parsing the json seems to be an artifact of SAS's specific technique for doing that, so that piece can be ignored.

**Comment by isaiah-r86** [comment]
I haven't encountered any files with this format, do you have a link to some examples?

**Reply by lkoll** [reply]
Not really sure if I can link to a file, but one file I am looking at with that structure is named this: 2022-07-01_UnitedHealthcare-Insurance-Company_Insurer_IND-N-A_N4_in-network-rates.json.gz

You should be able to ctrl f that or some subset of it to see.

Can I ask in what way is the structure different from what you've seen? And also would be curious for an example file name if you have one available.

Edit: I guess I should have asked, have you been looking at UHC files or other insurers? If you haven't been looking at UHC you would search for that filename at this link: https://transparency-in-coverage.uhc.com/undefined

**Reply by lkoll** [reply]
@isaiah-r86 , did you ever get a chance to look at that file?

**Reply by isaiah-r86** [reply]
Unfortunately that specific file is no longer up on the UHC site but I took a look at the [2022-08-01 version of the file](https://uhc-tic-mrf.azureedge.net/public-mrf/2022-08-01/2022-08-01_UnitedHealthcare-Insurance-Company_Insurer_IND-N-A_N4_in-network-rates.json.gz) using the [less command](https://en.wikipedia.org/wiki/Less_(Unix)) in a terminal and didn't have any issues.

**Reply by lkoll** [reply]
So when you parse the json file how are you separating it out different from the structure I mentioned above?

**Reply by isaiah-r86** [reply]
I just unzipped the file and opened it with `less` and it appears as a normally structured JSON file to me.  Is it possible that SAS creates this structure while parsing the file?

I'd suggest checking out discussion #537 for recommendations on parsing out these larger files.

**Reply by lkoll** [reply]
I am sorry if I was unclear, I realize the json file is normal. But once you parse the json file are you not breaking it into a structure similar to above for analysis? I'm confused what you're doing with the json in just json format.
